## 08_fix_ftbfs_on_hurd-i386 by HIGUCHI Daisuke (VDR dai) <dai@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix FTBFS on hurd-i386. use O_RDONLY instead of bare 0.

--- canna-3.7p3/cmd/crxdic/crxdic.c.orig	2003-12-27 17:15:21.000000000 +0000
+++ canna-3.7p3/cmd/crxdic/crxdic.c	2012-01-27 05:12:18.000000000 +0000
@@ -1228,7 +1228,7 @@ main (argc, argv)
     if (!(dic->gramdata = RkiReadWholeFile(fp, &dic->gramsz)))
       goto gram_err;
     fclose(fp);
-    if ((fd = open(gfile, 0)) < 0 || !(gram = RkReadGram(fd, dic->gramsz)))
+    if ((fd = open(gfile, O_RDONLY)) < 0 || !(gram = RkReadGram(fd, dic->gramsz)))
       goto gram_err;
     close(fd);
     goto gram_ok;
